### PR TITLE
add Makefile and pyInstaller build script

### PR DESCRIPTION
**Pull request by Rob1in**
follow-up pr is to add how to create the venv, etc...

### CODE REVIEW

**Review by penguinland — COMMENTED** [review]
I can't tell if my feedback is actionable. Maybe the answers to my questions are all "the PR should be left as-is with no changes." 

The big picture LGTM, but I'd at least like answers to the questions.

File: .gitignore [hunk 1]
```diff
@@ -12,10 +12,7 @@ results*
 #ignore models
 /src/models/
 
-#from pyinstaller
-/main/
-/build/
-/dist/main
+
```
[the remark this answers is not in the document]
Comment (Rob1in) [reply .gitignore]: done

File: README.md [hunk 2]
```diff
@@ -142,25 +142,49 @@ Recomputes embeddings.
...
+##### Usage:
+
+To build the project with the default paths:
+
+```bash
+make pyinstaller
 ```
 
-Make sure that the requirements are installed:
+To build the project with custom paths:
```
[the remark this answers is not in the document]
Comment (Rob1in) [reply README.md]: removed those

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,15 @@
+.PHONY: pyinstaller clean-pyinstaller
+
+# Default values for workpath and distpath
+WORKPATH ?= ./pyinstaller_build      # Default build output directory
+DISTPATH ?= ./pyinstaller_dist       # Default distribution output directory
```
[the remark this answers is not in the document]
Comment (Rob1in) [reply Makefile]: done

File: README.md [hunk 4]
```diff
@@ -142,25 +142,49 @@ Recomputes embeddings.
...
 
-Run this to create your virtual environment:
-Activate it bby running:
-```
-source .venv/bin/activate
+This project includes a `Makefile` and a `build_installer.sh` script to automate the PyInstaller build process. PyInstaller is used to create standalone executables from the Python module scripts.
+
+### Available Commands
+
+#### 1. `pyinstaller`
```
[the remark this answers is not in the document]
Comment (Rob1in) [reply README.md:151]: done

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+# The 'pyinstaller' target runs the build/build_pyinstaller.sh script
+pyinstaller:
+	WORKPATH=$(PYINSTALLER_WORKPATH) DISTPATH=$(PYINSTALLER_DISTPATH) ./build/build_pyinstaller.sh
+
+# The 'clean-pyinstaller' target removes the workpath and distpath directories
+clean-pyinstaller:
+	@echo "Cleaning PyInstaller workpath: $(PYINSTALLER_WORKPATH)"
+	@echo "Cleaning PyInstaller distpath: $(PYINSTALLER_DISTPATH)"
+	rm -rf $(PYINSTALLER_WORKPATH) $(PYINSTALLER_DISTPATH)
+
```
Comment (penguinland) [inline Makefile]: We talked in person and I think you've got VSCode configured correctly for next time. It's still off in this diff, but it's good enough for the long term.

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+# The 'pyinstaller' target runs the build/build_pyinstaller.sh script
+pyinstaller:
+	WORKPATH=$(WORKPATH) DISTPATH=$(DISTPATH) ./build/build_pyinstaller.sh
+
+# The 'clean-pyinstaller' target removes the workpath and distpath directories
+clean-pyinstaller:
+	@echo "Cleaning PyInstaller workpath: $(WORKPATH)"
+	@echo "Cleaning PyInstaller distpath: $(DISTPATH)"
+	rm -rf $(WORKPATH) $(DISTPATH)
```
[the remark this answers is not in the document]
Comment (Rob1in) [reply Makefile]: done

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+PYINSTALLER_DISTPATH = ./pyinstaller_dist
+
+# The 'pyinstaller' target runs the build/build_pyinstaller.sh script
+pyinstaller:
+	WORKPATH=$(PYINSTALLER_WORKPATH) DISTPATH=$(PYINSTALLER_DISTPATH) ./build/build_pyinstaller.sh
+
+# The 'clean-pyinstaller' target removes the workpath and distpath directories
+clean-pyinstaller:
+	@echo "Cleaning PyInstaller workpath: $(PYINSTALLER_WORKPATH)"
+	@echo "Cleaning PyInstaller distpath: $(PYINSTALLER_DISTPATH)"
```
Comment (penguinland) [inline Makefile:14]: I'm about to learn something! What does the `@` at the start of the line do?

**Review by penguinland — APPROVED** [review]
LGTM!

File: Makefile (hunk 5 above)
Comment (penguinland) [reply Makefile]: You're too fast! You've already fixed this. :wink:

**Pull request merged**